vmsa/vmsa.go

```diff
@@ -229,3 +236,68 @@ func (v VMSA) Pages(vcpus int) ([][]byte, error) {
 	}
 	return result, nil
 }
+
+// VMSA_SVSM represents the VMSA for Secure Virtual Machine Secure Monitor
+type VMSA_SVSM struct {
+	SaveArea SevEsSaveArea
+}
+
+func NewSVSM(apEip uint32, vcpuSig uint64, vmmType vmmtypes.VMMType) (VMSA_SVSM, error) {
+	sevFeatures := uint64(0x1)
+	saveArea, err := BuildSaveAreaSVSM(apEip, sevFeatures, vcpuSig, vmmType)
+	if err != nil {
+		return VMSA_SVSM{}, err
+	}
+	return VMSA_SVSM{SaveArea: saveArea}, nil
+}
+
+func BuildSaveAreaSVSM(eip uint32, sevFeatures uint64, vcpuSig uint64, vmmType vmmtypes.VMMType) (SevEsSaveArea, error) {
```

Comment:
```suggestion
func (VMSA_SVSM) BuildSaveArea(eip uint32, sevFeatures uint64, vcpuSig uint64, vmmType vmmtypes.VMMType) (SevEsSaveArea, error) {
```

the original `BuildSaveArea` should also be a method of `VMSA`. but I can do that in a separate PR.

Reply:
I'll leave that for another PR